packages/app-project/src/components/AuthModal/shared/components/withFormik/withFormik.spec.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import { expect } from 'chai'
+import { Box, Button, FormField, TextInput } from 'grommet'
+import { Formik } from 'formik'
+import sinon from 'sinon'
+import withFormik from './withFormik'
+
+const MockForm = ({ handleBlur, handleChange, handleSubmit, values }) => (
+  <Box as='form' onSubmit={handleSubmit}>
+    <FormField htmlFor='testInput' label='Enter text' >
+      <TextInput
+        id='testInput'
+        name='testInput'
+        onBlur={handleBlur}
+        onChange={handleChange}
+        type='text'
+        value={values.testInput}
+      />
+    </FormField>
+    <Button type='submit' label='Submit' />
+  </Box>
+)
+
+const WrappedMockForm = withFormik(MockForm)
+
+const initialValues = { testInput: '' }
+const onSubmit = sinon.spy()
+
+describe('Higher Order Component > withFormik', function () {
+  it('renders without crashing', function () {
+    const wrapper = shallow(<WrappedMockForm />)
+    expect(wrapper).to.be.ok()
+  })
+
+  it('should pass Formik required props', function () {
+    const wrapper = shallow(
+      <WrappedMockForm
+        initialValues={initialValues}
+        onSubmit={onSubmit}
+      />
+    )
+
+    const formikComponent = wrapper.find(Formik)
+
+    expect(formikComponent.props().initialValues).to.equal(initialValues)
+    expect(formikComponent.props().onSubmit).to.equal(onSubmit)
+  })
+
+  it('should pass Formik optional props', function () {
+    const validate = sinon.spy()
+    const wrapper = shallow(
+      <WrappedMockForm
+        initialValues={initialValues}
+        onSubmit={onSubmit}
+        validate={validate}
+      />
+    )
+    const formikComponent = wrapper.find(Formik)
+
+    expect(formikComponent.props().validate).to.equal(validate)
+  })
+
+  it('should call onChange prop', function () {
+    const onChangeStub = sinon.stub()
+    const handleChangeStub = sinon.stub()
+    const eventMock = {}
+    const formikPropsMock = {}
+    const wrapper = shallow(
+      <WrappedMockForm
+        initialValues={initialValues}
+        onChange={onChangeStub}
+        onSubmit={onSubmit}
+      />
+    )
+
+    wrapper.instance().handleChangeWithCallback(eventMock, handleChangeStub, formikPropsMock)
```

Comment:
If you can call `WrappedComponent.props.handleChange` here, that would be a more robust test.

Reply:
It would be in theory, but the `WrappedComponent` is the return value for Formik's render. Formik internals are hard to work with. Some async actions are happening which makes it hard to test without using a timeout, and I think an expectation inside a timeout in this case is potentially unreliable. Maybe this will change in the v2 release? If you have an idea to make this less brittle, I'm open to suggestion.

Reply:
Sorry, I missed this on my first pass, but the mock Formik props here should not be empty but should mock the expected inner props for these values of `initialValues`, `onSubmit` and `onChange`.

Reply:
This afternoon, I experimented with using a named function to get the wrapped component in tests but couldn't get that to work. Refs might be the way to go if you want to test the props that are passed down to the wrapped form.